src/renderers/shaders/ShaderChunk/clipping_planes_vertex.glsl

```diff
@@ -0,0 +1,4 @@
+#if NUM_USER_CLIPPING_PLANES > 0 && ! defined( STANDARD ) && ! defined( PHONG )
+	vViewPosition = - mvPosition.xyz;
```

Comment:
I think that Three.js should undergo a similar transition as graphics APIs did when switching from fixed-function pipelines to programmable ones at some point in the (hopefully) not so distant future. Putting one clever patch on top of the other won't make it any less patchy and will only sustain our suffering. 

Thinking in terms of dependencies is certainly the right direction. Let there be data structures, generic data-driven machinery and well-defined interfaces for extension!
Scanning belongs into a build step, IMO. This way, it doesn't have to be super lightweight and can be done properly without bloating up the library core.

A dependency evaluator capable of piecing together programs from appropriately prepared GLSL parts on-demand (handling arbitrary graphs, including automatic synthesis of copy operations on forking connections, ordering constraints and error checking) can be implemented in just a few hundred bytes of JavaScript. A generic mechanism that determines the data sources for uniforms (object, material, traversal rule, ...) takes even less code. It all boils down to a few, well thought-out, yet in the end much simpler and easier to maintain key components that let us turn all the grown mess into data.

Retiring the macros and stuff  in `WebGLProgram(s)` plus resulting simplification of the renderer may easily free-up the space for it all - and the library will become exponentially more powerful. It should be possible for the user to hook into the default shading system, rewire or partially override it at any point between the blocks - instead of having to go through a lengthy and error-prone process of copy & paste and reverse-engineering of the glorious ShaderChunk framework.

Stuff for another PR. You know I love that topic.